feature-libs/user/account/components/guards/login-as-guest.guard.ts

```diff
@@ -31,9 +33,12 @@ export class LoginAsGuestGuard {
     ) {
       this.windowRef.localStorage.removeItem(IS_GUEST_USER_CHECKOUT_KEY);
       return of(
-        this.router.createUrlTree([this.semanticPathService.get('loginForm')], {
-          queryParams: { forced: true },
-        })
+        this.router.createUrlTree(
+          [this.semanticPathService.get(this.routeName)],
+          {
+            queryParams: { forced: true },
+          }
+        )
```

Comment:
`major:` as mentioned, let's change the simple field to method, for better customisability (in case someone requires more complex logic to get the proper path). Changing the severity to 'major`. 
```suggestion
        this.router.createUrlTree(
          [this.getPath()],
          {
            queryParams: { forced: true },
          }
        )
        
        /* ... */
        
        protected getPath() {
           this.semanticPathService.get('loginForm');
        }
```

Reply:
done